cli/src/commands/tag.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020-2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use jj_lib::str_util::StringPattern;
+
+use crate::cli_util::{parse_string_pattern, CommandError, CommandHelper};
+use crate::ui::Ui;
+
+/// Manage tags.
+#[derive(clap::Subcommand, Clone, Debug)]
+pub enum TagSubcommand {
+    #[command(visible_alias("l"))]
+    List(TagListArgs),
+}
+
+/// List tags.
+#[derive(clap::Args, Clone, Debug)]
+pub struct TagListArgs {
+    /// Show tags whose local name matches
+    ///
+    /// By default, the specified name matches exactly. Use `glob:` prefix to
+    /// select tags by wildcard pattern. For details, see
+    /// https://github.com/martinvonz/jj/blob/main/docs/revsets.md#string-patterns.
+    #[arg(value_parser = parse_string_pattern)]
+    pub names: Vec<StringPattern>,
+}
+
+pub fn cmd_tag(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &TagSubcommand,
+) -> Result<(), CommandError> {
+    match subcommand {
+        TagSubcommand::List(sub_args) => cmd_tag_list(ui, command, sub_args),
+    }
+}
+
+fn cmd_tag_list(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &TagListArgs,
+) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+    let repo = workspace_command.repo();
+    let view = repo.view();
+
+    ui.request_pager();
+    let mut formatter = ui.stdout_formatter();
+    let formatter = formatter.as_mut();
+
+    for (name, _) in view.tags().iter() {
```

Comment:
nit: `.iter()` can be omitted.

`.keys()` can be used instead, but we'll need the ref target anyway to print a tagged commit like `jj branch list`.